bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ValueUtils.java

```diff
@@ -59,17 +74,118 @@ public class ValueUtils {
     public static BMap<BString, Object> createRecordValue(Module packageId, String recordTypeName) {
         ValueCreator valueCreator = ValueCreator.getValueCreator(ValueCreator.getLookupKey(packageId, false));
         try {
-            return valueCreator.createRecordValue(recordTypeName);
+            return getPopulatedRecordValue(valueCreator, recordTypeName);
         } catch (BError e) {
             // If record type definition not found, get it from test module.
             String testLookupKey = ValueCreator.getLookupKey(packageId, true);
             if (ValueCreator.containsValueCreator(testLookupKey)) {
-                return ValueCreator.getValueCreator(testLookupKey).createRecordValue(recordTypeName);
+                return getPopulatedRecordValue(ValueCreator.getValueCreator(testLookupKey), recordTypeName);
```

Comment:
This is because those lines are for record value creation with no strand. The actual use case is covered in std libs like - HTTP, SQL. 
I'll add tests using interop functions to recreate the scenario

Reply:
Ack